apps/src/dance/lab2/views/DanceView.tsx

```diff
@@ -302,6 +280,7 @@ const DanceView: React.FunctionComponent<{
   // Reset hasRun and hasEdited flag when level changes
   useEffect(() => {
     dispatch(setHasRun(false));
+    dispatch(setLab2HasRun(false));
```

Comment:
`hasRun` already exists in the DanceView, so to avoid doing a big refactor, I'm going to create a duplicate state.